tests/collections/vlm/hf/peft.py

```diff
@@ -0,0 +1,128 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import fiddle as fdl
+import torch
+from lightning.pytorch.loggers import WandbLogger
+
+from nemo import lightning as nl
+from nemo.collections import llm, vlm
+
+DATA_PATH = "/home/TestData/vlm/rdr-items"
+
+
+def mk_hf_vlm_dataset(processor, mbs, gbs):
+    skipped_tokens = vlm.HFAutoModelForImageTextToText.extract_skipped_token_ids(processor)
+
+    def collate_fn(examples, processor):
+        def fmt(sample):
+            instruction = "Describe accurately the given image."
+            conversation = [
+                {
+                    "role": "user",
+                    "content": [{"type": "text", "text": instruction}, {"type": "image", "image": sample["image"]}],
+                },
+                {"role": "assistant", "content": [{"type": "text", "text": sample["text"]}]},
+            ]
+            return {"conversation": conversation, "images": [sample['image']]}
+
+        text = []
+        images = []
+        for example in map(fmt, examples):
+            text.append(
+                processor.apply_chat_template(
+                    example["conversation"],
+                    tokenize=False,
+                    add_generation_prompt=False,
+                )
+            )
+            images += example['images']
+
+        # Tokenize the text and process the images
+        batch = processor(
+            text=text,
+            images=images,
+            padding=True,
+            return_tensors="pt",
+        )
+
+        batch["pixel_values"] = batch["pixel_values"].to(torch.bfloat16)
+
+        labels = batch["input_ids"].clone()
+        labels[torch.isin(labels, skipped_tokens)] = -100
+        batch["labels"] = labels
+        return batch
+
+    return vlm.HFDatasetDataModule(
+        DATA_PATH,
+        split="train[:10]",
```

Comment:
what does [:10] mean here?

Reply:
just use the first ten items, similar to python list operator